common/utils/random.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"github.com/DataDog/test-infra-definitions/common/namer"
+	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type RandomGenerator struct {
+	ctx      *pulumi.Context
+	namer    namer.Namer
+	provider *random.Provider
+}
+
+func WithProvider(provider *random.Provider) func(*RandomGenerator) {
+	return func(r *RandomGenerator) {
+		r.provider = provider
+	}
+}
+
+func NewRandom(ctx *pulumi.Context, options ...func(*RandomGenerator)) (*RandomGenerator, error) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
Struct constructor should match struct name - better replace from your IDE than apply this suggestion, it will break the linting

```suggestion
func NewRandomGenerator(ctx *pulumi.Context, options ...func(*RandomGenerator)) (*RandomGenerator, error) {
```